### PR TITLE
feat: update block perf tests to use fio json

### PR DESCRIPTION
**Pull request by ShadowCurse**
## Changes
Block tests were using fio logs to calculate bandwidth. The issue is that the values in logs are in KB/sec and we were just summing them up, so getting a sum of averages. But what we want is just an average bandwidth for the test.
Now we use json output of fio which contains much more info in a nice format. This info already includes all bandwidth calculations. This change will also allow us to output other metrics in the future if we want.

## Reason
More correct block performance tests

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this
  PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/integration_tests/performance/test_block_ab.py [hunk 1]
```diff
@@ -96,38 +96,23 @@ def run_fio(microvm, mode, block_size):
         assert rc == 0, stderr
         assert stderr == ""
 
-        microvm.ssh.scp_get("/tmp/*.log", logs_path)
-        microvm.ssh.check_output("rm /tmp/*.log")
+        microvm.ssh.scp_get("/tmp/fio.json", output_path)
+        microvm.ssh.check_output("rm /tmp/fio.json")
 
-        return logs_path, cpu_load_future.result()
+        return output_path, cpu_load_future.result()
```
Comment (pb8o) [inline tests/integration_tests/performance/test_block_ab.py:102]: `return output_path / "fio.json", ...`

File: tests/integration_tests/performance/test_block_ab.py [hunk 2]
```diff
@@ -96,38 +96,23 @@ def run_fio(microvm, mode, block_size):
...
 
 
-def process_fio_logs(vm, fio_mode, logs_dir, metrics):
-    """Parses the fio logs in `{logs_dir}/{fio_mode}_bw.*.log and emits their contents as CloudWatch metrics"""
+def process_fio_json(json_path, metrics):
+    """Parse fio json ouput"""
+    with open(os.path.join(json_path, "fio.json"), "r", encoding="utf-8") as file:
+        j = json.load(file)
```
Comment (pb8o) [inline tests/integration_tests/performance/test_block_ab.py:108]: Combined with above comment:

```python
fio = json.loads(json_path.read_text(encoding="utf-8") 
```

File: tests/integration_tests/performance/test_block_ab.py (hunk 1 above)
Comment (ShadowCurse) [reply tests/integration_tests/performance/test_block_ab.py]: done

File: tests/integration_tests/performance/test_block_ab.py (hunk 2 above)
Comment (ShadowCurse) [reply tests/integration_tests/performance/test_block_ab.py]: done

**Comment by ShadowCurse** [comment]
Each line in the log contains `value seen over the specified period of time` (I assume this is an average over 1s because we specify to log every second). So each line contains KB/s (not just KB). We were summing these KB/s and were getting a `sum of averages`.
And if they were KB, then our tests were completely wrong, as were calculating the total amount of data send instead of the bandwidth.

**Review by roypat — CHANGES_REQUESTED** [review]
Ah, I see now. This won't work with our A/B-tests, because they require a data series to be emitted, not just a single data point.

I admittedly still don't quite understand what was wrong with the old approach. It gives you a data series where each entry is "number of bytes written/read since the last entry", with entries being emitted once per second. That's different from "kb/s", because its not "average bandwidth since the test started". So it's fine to sum up the data points at each second (and if it weren't, I don't see why your summing up of _all_ datapoints wouldn't be just as problematic).

**Comment by ShadowCurse** [comment]
I am pretty sure A/B works with 1 data point as this is what is happening now with VMs with 1 vcpu (1 fio job). We sum all values and emit 1 metric.

**Pull request closed without merging**